Library/Homebrew/env_config.rb

```diff
@@ -504,5 +505,22 @@ def cask_opts_require_sha?
     def automatically_set_no_install_from_api?
       ENV["HOMEBREW_AUTOMATICALLY_SET_NO_INSTALL_FROM_API"].present?
     end
+
+    sig { returns(T::Boolean) }
+    def cask_upgrade_greedy?
+      ENV["HOMEBREW_UPGRADE_GREEDY"].present? &&
+        !cask_upgrade_greedy_latest? &&
+        !cask_upgrade_greedy_auto_updates?
```

Comment:
Feels a little weird/loose for "any value except `latest` or `auto-updates` implies greedy". Maybe we should only allow this value to be set to `1` or `greedy` or some few known values and error out otherwise?

Reply:
Previously `HOMEBREW_UPGRADE_GREEDY` was a Boolean so accepting any value is to maintain compatibility with any existing value a user may have set (e.g. "1", "true", "yes", etc).

However I agree that it would be nicer to accept specific values (e.g. "all", "latest", and "auto-updates") with some error handling to warn about invalid values and help users migrate from previously valid values (a.k.a. "1", etc).